mvn-extension/src/main/java/org/arquillian/smart/testing/mvn/ext/MavenProjectConfigurator.java

```diff
@@ -47,11 +46,33 @@ void addRequiredDependencies(Model model) {
                 final Dependency dependency = dependencies.get(strategy);
                 model.addDependency(dependency);
             }
+            addSurefireApiDependency(model);
         } catch (IOException e) {
             throw new RuntimeException("Unable to load strategy definitions", e);
         }
     }
 
+    private void addSurefireApiDependency(Model model) {
+        if (surefireVersion != null && surefireVersion.toString() != null) {
+            boolean alreadyContains = model.getDependencies().stream()
+                .filter(dep ->
+                    dep.getGroupId().equals("org.apache.maven.surefire") && dep.getArtifactId().equals("surefire-api"))
+                .findFirst().isPresent();
```

Comment:
Use `anyMatch` so you don't need to get the `Optional` and the pass with `isPresent`